src/OpenRiaServices.Tools/Framework/MetadataPipeline/StandardCustomAttributeBuilder.cs

```diff
@@ -103,8 +103,14 @@ protected virtual string MapProperty(PropertyInfo propertyInfo, Attribute attrib
         {
             string propertyName = propertyInfo.Name;
 
-            // We strip "TypeId" from all Attributes.  It is a property we cannot set via setters or ctor
-            return propertyName.Equals("TypeId", StringComparison.Ordinal) ? null : propertyName;
+            if (propertyName.Equals("TypeId", StringComparison.Ordinal))
+                return null;
+#if NET6_0_OR_GREATER
+            // We strip "TypeId" from all CustomValidationAttribute for NET 6.0 since we cannot set it, it is calculated indirectly
+            else if (attribute is CustomValidationAttribute && propertyInfo.Name == nameof(CustomValidationAttribute.RequiresValidationContext))
```

Comment:
Why is RequiresValidationContext removed ?
I had a look and does not BuildPropertyMaps handle this case?

- It might be possible to have the code in the dedicated "CustomValidationCustomAttributeBuilder" class instead.

It seems strange that limitations on server runtime should affect generated client code (but it might be another issue). Ideally only client project and runtime should affect which properties are generated or not

Reply:
I've updated the comment. 
While it can be solved in CustomValidationCustomAttributeBuilder, that solution has a slightly risk of breaking other (non system) ValidationAttributes having the property set to true